ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PreJoinRepartitionNode.java

```diff
@@ -111,7 +112,7 @@ public void setKeyFormat(final KeyFormat format) {
     return getSource().buildStream(builder)
         .selectKey(
             valueFormat.getFormatInfo(),
-            partitionBy,
+            Collections.singletonList(partitionBy),
```

Comment:
nit: same thing about `ImmutableList`

Reply:
Sure. For my edification, why is ImmutableList preferred?